Document Approval on RFC-674

gcomoretto opened the pull request.

gcomoretto committed:
First Empty Draft

jonathansick committed:
Add Travis CI deployment to ldm-732.lsst.io

jonathansick committed:
Update deployment credentials

gcomoretto committed:
Test Cases included in draft

gcomoretto committed:
Fix order

gcomoretto committed:
Fixed formatting issue

gcomoretto committed:
Trimmed pandoc strings

gcomoretto committed:
Added traceability

gcomoretto committed:
Added Introduction on JK input

gcomoretto committed:
Rephrase

gcomoretto committed:
Update changelog

gcomoretto committed:
Merge branch 'tickets/DM-23219'

gcomoretto committed:
Updated title

gcomoretto committed:
Fixed traceability table

gcomoretto committed:
Added new technical report

gcomoretto committed:
Fix .travis.yml

gcomoretto committed:
Added missing date

gcomoretto committed:
Review as per Leanes comments in RFC-674

gcomoretto committed:
Regenerate from Jira

gcomoretto committed:
Added conf.py

gcomoretto committed:
Regenerate from Jira

gcomoretto committed:
Updated from Jira

gcomoretto committed:
Updated details template and regenerated from Jira

gcomoretto committed:
Added section introduction and scope

gcomoretto committed:
Added document-14789 to local.bib

leannep commented on ve_details.tex:
```diff
@@ -0,0 +1,2698 @@
+  \href{https://docushare.lsstcorp.org/docushare/dsweb/Get/Document-35302/DAQ\%20DWDM\%20connection\%20tests\%2020191109.pptx}{DAQ
+  DWDM Connection Tests})
+\end{enumerate}
+
```
This is not an objective but rather a description of how the test will be carried out.  The test objective should define the overall goal and achievement of the test execution. 

E.g  "Verify the integration of the summit to base network by demonstrating a sustained and uninterrupted transfer of  data between summit and base over 1 day period at or exceeding  rates specified in
  \citeds{LDM-142}."

The phases should be moved into either the test procedure or should be separate test cases that must be executed to verify this requirement

leannep commented on ve_details.tex:
```diff
@@ -0,0 +1,2698 @@
+specification. Network operator will provide MTTR data on links during
+commissioning and operations.\\[2\baselineskip]
+
+\textbf{Precondition:} \\
```
Objective reads more like a test case. Rephrase to something like: 
"Verify reliability of summit to base network by demonstrating reconnection and recovery to
transfer of data at or exceeding rates specified in \citeds{LDM-142} following a cut in network. 

Remaining details should be moved to the test case procedure.

leannep commented on ve_details.tex:
```diff
@@ -0,0 +1,2698 @@
+Analyze the network and show that data acquired by a DAQ can be
+transferred within the required time, i.e. verify that link is capable
+of transferring image for prompt processing in oArchiveMaxTransferTime =
+5{[}second{]}, i.e. at or exceeding rates specified in \citeds{LDM-142}.
```
Simplify to:
Verify that the base to archive network is capable
of transferring images acquired for prompt processing at or exceeding rates specified in \citeds{LDM-142}, i.e in oArchiveMaxTransferTime = 5{[}second{]}."

leannep commented on ve_details.tex:
```diff
@@ -0,0 +1,2698 @@
+Via monitoring and ~comparing with at least 6 months of historical data
+on the link, verify that the mean time between failures is less than
+summToBaseNetMTBF (90 days) over 1 year.
+
```
"Verify the availability of Summit to Base Network by demonstrating that the mean time between failures is less than summToBaseNetMTBF (90 days) over 1 year."

Move the details (Via monitoring and ~comparing with at least 6 months of historical data
on the link) into the test case procedure.

leannep commented on ve_details.tex:
```diff
@@ -0,0 +1,2698 @@
+\textbf{Objective:} \\
+After disconnecting fiber at a location in the Base - Archive, network,
+verify recovery can occur within baseToArchNetMTTR = 48{[}hour{]}.
+
```
"Verify Base to Archive Network Reliability by demonstrating that the network can recover from outages within baseToArchNetMTTR = 48{[}hour{]}."

Details about how the outage is simulated (i.e After disconnecting fiber at a location in the Base - Archive, network)  should be in the test case procedure.

leannep commented on ve_details.tex:
```diff
@@ -0,0 +1,2698 @@
+capacity on secondary link, restore primary link and verify that network
+recovers to primary. Demonstrate use of secondary path in ``catch-up''
+mode.
+
```
"Verify Base to Archive Network Secondary Link failover and capacity, and subsequent recovery primary. Demonstrate use of secondary path in ``catch-up'' mode."

Details of how, e.g by simulations to go to the test case procedure.

leannep commented on ve_details.tex:
```diff
@@ -0,0 +1,2698 @@
+Monitor and analyze data transfers to verify networks can transfer data
+at archToDacBandwidth = 10000{[}megabit per second{]}, i.e. at or
+exceeding rates specified in \citeds{LDM-142}.
+
```
Verify archiving of data to  Data Access Center Network at or exceeding rates specified in \citeds{LDM-142}, i.e at archToDacBandwidth = 10000{[}megabit per second{]},.

Montitoring/analysis details to go to the test case procedure

leannep commented on ve_details.tex:
```diff
@@ -0,0 +1,2698 @@
+\textbf{Objective:} \\
+From test and historical data, extrapolate to 1 year to estimate
+networks meet archToDacNetMTBF = 180{[}day{]}.
+
```
Verify availability of archiving to Data Access Center Network using test and historical data of or exceedding archToDacNetMTBF= 180{[}day{]).

leannep commented on ve_details.tex:
```diff
@@ -0,0 +1,2698 @@
+~After reconnecting primary path, recovery of transfer of data on
+primary path, and verify recovery can meet chToDacNetMTTR =
+48{[}hour{]}.
+
```
Verify reliability of Archive to Data Access Center Network by demonstrating successful failover and capacity to the secondary part and  subsequent recovery to primary within or exceeding chToDacNetMTTR =
48{[}hour{]}.

leannep commented on ve_details.tex:
```diff
@@ -0,0 +1,2698 @@
+\textbf{Objective:} \\
+Inspect construction and operations contracts and Indefeasible Rights to
+Use (IRUs).
+
```
Verify Summit to Base Network Ownership and Operation by LSST and/or the operations entity by inspection of  operations contracts and Indefeasible Rights

leannep commented on ve_details.tex:
```diff
@@ -0,0 +1,2698 @@
+
+\textbf{Objective:} \\
+After simulating failure on primary path, verify fail-over to and
+capacity on secondary path.
```
Verify Archive to Data Access Center Network via Secondary Link by simulating a failure on primary path and capacity on the secondary path.

leannep commented on ve_details.tex:
```diff
@@ -0,0 +1,2698 @@
+expect to meet baseToArchNetMTBF = 180{[}day{]}. ~Note that this is for
+complete loss of transfer service (simultaneous failure on all paths),
+not a single path failure with successful fail-over.
+
```
Verify the availability of the Base to Archive Network communications by demonstrating that it  meets or exceeds a mean time between failures, measured over a 1-yr period of   MTBF > baseToArchNetMTBF (180[day])

leannep reviewed (changes requested):
Main comment is that many test case objectives contain implementation details. These details should be in the test case procedure and the objective should provide a clear statement of the goal of the test case. I have provided suggestions for rephrasing - feel free to improve on them.

gcomoretto committed:
Update from Jira

gcomoretto committed:
Fixed template

gcomoretto committed:
Document finalized

gcomoretto committed:
Removed draft

gcomoretto committed:
Added 1.0 version

gcomoretto committed:
Revert "Removed draft"

This reverts commit b815432abfece0a23ede674c71ecc842c8fad697.

gcomoretto committed:
Merge branch 'tickets/RFC-674'

jonathansick committed:
Convert build to use GitHub Actions

This removes the original Travis CI-based build, and migrates the build
to use GitHub Actions and Python 3.7.

jonathansick committed:
Merge pull request #3 from lsst/tickets/DM-24173

DM-24713: Convert build to use GitHub Actions

womullan committed:
add aciton

womullan committed:
fix docgen doc

womullan committed:
Merge pull request #4 from lsst/u/womullan/fixdocgen

Add action

womullan committed:
P47 >>

womullan committed:
Update from JIRA.

womullan committed:
history_and_info.tex

womullan committed:
Merge branch 'tickets/RFC-674' of github.com:lsst/LDM-732 into tickets/RFC-674

womullan committed:
fixed up for docgen

womullan committed:
update make

womullan committed:
works again

womullan committed:
the commdn line

womullan committed:
fix gen

womullan commented:
This RFC was implemtented.. ages ago ...
